HomeTask/HomeTask/Model/CoreDataManager.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  TasksPersistence.swift
+//  HomeTask
+//
+//  Created by Lucas Migge de Barros on 11/01/23.
+//
+
+import Foundation
+import CoreData
+
+
+protocol TaskPersistence {
+    func fetchTaks(completion: @escaping ([TaskItem]) -> Void)
+    
+    func saveTask(taskTitle: String, completion: @escaping () -> Void)
+    
+    func deleteTask(forTask task: TaskItem, completion: @escaping () -> Void)
+}
+
+
+class CoreDataManager: TaskPersistence {
```

Comment:
A ideia é que essa classe gerencie exclusivamente a persistência de Tasks, para evitar que ela esteja lidando com múltiplas responsabilidades, certo?
Nesse caso, seria interessante deixar isso claro em seu nome. `CoreDataManager` é bastante genérico e pode significar muitas coisas. Importante sempre escrevermos pensando em quão fácil esse código vai ser de outras pessoas entenderem no futuro, mesmo sem muito contexto.

Um exemplo de nome poderia ser:
```suggestion
class CoreDataTaskPersistence: TaskPersistence {
```
Deixa claro que está lidando com Tasks e da a entender que é uma implementação de `TaskPersistence` usando `CoreData`.

Reply:
Perfeito!